### PR TITLE
Update hooks to make more plugins works properly with PCT

### PR DESCRIPTION
**Pull request by bmunozm**
This PR is about making plugins working with the PCT that needs hooks, special tags and so on.

Fixes included:

- `jenkins-design-language` module in blueocean
- `github` plugin
- `pipeline-rest-api`

### CODE REVIEW

File: plugins-compat-tester-model/src/main/java/org/jenkins/tools/test/model/PomData.java [hunk 1]
```diff
@@ -45,8 +46,9 @@
     private String connectionUrl;
     private List<String> warningMessages = new ArrayList<String>();
 
-    public PomData(String artifactId, @CheckForNull String packaging, String connectionUrl, @CheckForNull MavenCoordinates parent){
+    public PomData(String artifactId, @CheckForNull String packaging, String connectionUrl, @CheckForNull MavenCoordinates parent, String groupId){
```
Comment (oleg-nenashev) [inline plugins-compat-tester-model/src/main/java/org/jenkins/tools/test/model/PomData.java]: Retain binary compatibility?

Comment (varyvol) [reply plugins-compat-tester-model/src/main/java/org/jenkins/tools/test/model/PomData.java]: I don't think it's used outside? But worth to verify that.

Comment (bmunozm) [reply plugins-compat-tester-model/src/main/java/org/jenkins/tools/test/model/PomData.java]: It is only used in this repo, so I guess it is fine.

File: plugins-compat-tester/src/main/java/org/jenkins/tools/test/hook/PipelineRestApiHook.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    protected String getPluginFolderName(UpdateSite.Plugin currentPlugin){
+        return "rest-api";
+    }
+
+    @Override
+    public boolean check(Map<String, Object> info) throws Exception {
+        return isPSVPlugin(info);
+    }
+
+    public static boolean isPSVPlugin(Map<String, Object> moreInfo) {
```
[the remark this answers is not in the document]
Comment (bmunozm) [reply plugins-compat-tester/src/main/java/org/jenkins/tools/test/hook/PipelineRestApiHook.java]: changed @varyvol

[the remark this answers is not in the document]
Comment (bmunozm) [reply plugins-compat-tester/src/main/java/org/jenkins/tools/test/hook/PipelineRestApiHook.java]: changed!

**Comment by raul-arabaolaza** [comment]
> `BlueOcean` filter is too aggressive IMO. it will match all components with BO in the group ID, Could we limit it to a JDL case only?

I believe that is already solved, now checks also the artifactId, here we can be less or more agressive, in one case we may run the hook for a wrong plugin, in another we may not run the hook for a needed plugin... Now we are in the second position which I believe is better and we will not break previously working plugins so approving it

File: plugins-compat-tester-model/src/main/java/org/jenkins/tools/test/model/PomData.java (hunk 1 above)
Comment (oleg-nenashev) [reply plugins-compat-tester-model/src/main/java/org/jenkins/tools/test/model/PomData.java]: We can live with that

**Pull request merged**